### DISCUSSION TITLE
Additional fields to the signature pad submission

### DISCUSSION THREAD

**Discussion by GreggBarfield** [szimek/signature_pad#727] [Q&A]
Is there a way to add addtitional fields to the signature pad submission and make them 'required'?

**Comment by UziTech** [comment]
That depends on how you submit the signature. This library only handles writing on the canvas and creating the image. How you submit it and other fields is up to you.

**Reply by GreggBarfield** [reply]
I can submit the other fields as part of the form; no problem.     I need some javascript (or JQuery) that goes in here that allows me to validate the other fields (make them REQUIRED)  as well.    I'm surprised this has never been asked as many times other fields are submitted in forms along with the signature.:

savePNGButton.addEventListener("click", function (event) {
  if (signaturePad.isEmpty()) {
    alert("Please provide a signature first.");
  } else {
    input.value = signaturePad.toDataURL();
        form.submit();
  }
});

**Reply by UziTech** [reply]
I'm sure it has been asked, just not in this repo, because this library doesn't submit anything.
